### ISSUE TITLE
Continued Maintenance

### ISSUE THREAD

**Issue by kyleboe** [absinthe-graphql/absinthe-socket#66]
I fully acknowledge that the last few years have been a roller coaster so I bring this up with all the empathy I can. There are several valuable PRs that have been opened over the last couple years without much input from those currently maintaining this repo. I would be happy to help with maintenance if the need is there. Ultimately, I want to do my part to ensure that the OSS on which I depend does not die on the vine.

TL;DR Thoughts on adding maintainers?

**Comment by kyleboe** [comment]
Thanks so much for the context. Admittedly, I don't have the strongest JS chops so I don't know that I am qualified enough to maintain this. However, if there is nobody else available, I am willing to assist. 


> Notably though there is an alternative route that I think may be more sustainable: providing a Phoenix Websocket implementation that is compatible with Apollo Server's wire protocol. This would allow front end developers to simply use the standard, off the shelf apollo-link and go from there.
>
> I believe there has already been some work done on this, but its status is unknown to me at the moment.

Interesting, that definitely seems to be a more pragmatic approach. I will poke around the Apollo docs and see what can be done to this effect.

**Comment by benwilson512** [comment]
https://github.com/easco/absinthe_apollo_sockets is one implementation. It looks like overall the apollo socket protocol is a lot simpler than phoenix channels. Probably the main concern I have is how testing it will go, as I'd like to do that end to end, and that introduces a need to keep a local apollo project up to date.

**Comment by kyleboe** [comment]
It looks like the library linked to in the [README of `easco/absinthe_apollo_sockets`](https://github.com/easco/absinthe_apollo_sockets/blob/master/README.md) is no longer maintained (see: [here](https://github.com/apollographql/subscriptions-transport-ws/blob/master/README.md)). The [updated Apollo docs](https://www.apollographql.com/docs/react/api/link/apollo-link-subscriptions/) also recommend using the [`graphql-ws` standard/library](https://github.com/enisdenjo/graphql-ws). Maybe I am missing something, but could solely `graphql-ws` be used client-side?

**Comment by maartenvanvliet** [comment]
I recommend https://github.com/geometerio/absinthe_graphql_ws for using the `graphql-ws` protocol. I think by now all Apollo clients (iOS/Kotlin/web) should have support for it. 

One thing I've been meaning to look into is whether it's possible to do some kind of protocol negotiation. Then you could point the client at the socket and the backend would figure out which protocol is intended and use that.

**Comment by kyleboe** [comment]
@maartenvanvliet I think we are talking about the JS (client) side of things. Not necessarily consuming GQL via the backend (elixir/absinthe). Maybe I'm misunderstanding your recommendation though.

**Comment by benwilson512** [comment]
@kyleboe It's actually super relevant. What @maartenvanvliet and I are saying is that instead of maintaining an Absinthe specific JS client, what we want to do is provide official Absinthe support for the _normal standard js client_ so that our relative lack of JS developers stops being a problem.

**Comment by kyleboe** [comment]
Ah I see! I misread the Usage section of the `absinthe_graphql_ws` readme. That makes total sense. Seems like a great route to take!

**Comment by kyleboe** [comment]
Ok yeah I just got everything running locally by implementing [`absinthe_graphql_ws`](https://github.com/geometerio/absinthe_graphql_ws) and removing `@absinthe/socket`/`@absinthe/socket-apollo-link` dependency as well as the JS [`phoenix` dependency](https://www.npmjs.com/package/phoenix).

I basically followed the [Apollo docs here](https://www.apollographql.com/docs/react/api/link/apollo-link-subscriptions) and ensured that the backend was set up as outlined in the [`absinthe_graphql_ws` docs](https://hexdocs.pm/absinthe_graphql_ws/installation.html#configuration) and it Just Worked:tm:. 

Thanks so much for all the help @benwilson512 & @maartenvanvliet! 

FWIW

> what we want to do is provide official Absinthe support on the Elixir side for the normal standard js client so that our relative lack of JS developers stops being a problem

This seems like a great way to go. If there are any PRs I can open to assist in this just let me know.

**Comment by enewbury** [comment]
Also had success using this library. Perhaps it would be valuable to add a section in the absinthe docs showcasing this as an alternative for folks who need to use Apollo V3
